Sources/ApodiniNetworking/HTTP/Request.swift

```diff
@@ -0,0 +1,161 @@
+    struct ParametersStorage: Hashable {
+        /// The requst's parameters that are expressed via a clear key-value mapping (e.g.: explicitly named path parameters)
+        var namedParameters: [String: String] = [:]
+        /// The request's parameters that are the result from matching the request againat a ``HTTPRouter.Route`` which contained single-path-component wildcards
+        /// key: index of wildcard in path components array
+        var singleComponentWildcards: [Int: String] = [:]
+        /// The request's parameters that are the result from matching the request againat a ``HTTPRouter.Route`` which contained multi-path-component wildcards
+        /// key: index of wildcard in path components array
+        var multipleComponentWildcards: [Int: [String]] = [:]
+    }
+    
+    
+    public let remoteAddress: SocketAddress?
+    public let version: HTTPVersion
+    public let method: HTTPMethod
+    public let url: URI
+    public var headers: HTTPHeaders
+    public var bodyStorage: BodyStorage
+    public let eventLoop: EventLoop
+    
+    /// For incoming requests from external clients processed through the HTTP server's router, the route this request matched against.
+    /// - Note: This property is `nil` for manually constructed requests
+    internal var route: HTTPRouter.Route?
+    
+    private var parameters = ParametersStorage()
+    
+    
+    public init(
+        remoteAddress: SocketAddress? = nil,
+        version: HTTPVersion = .http1_1,
+        method: HTTPMethod,
+        url: URI,
+        headers: HTTPHeaders = [:],
+        bodyStorage: BodyStorage = .buffer(),
+        eventLoop: EventLoop
+    ) {
+        self.remoteAddress = remoteAddress
+        self.version = version
+        self.method = method
+        self.url = url
+        self.headers = headers
+        self.bodyStorage = bodyStorage
+        self.eventLoop = eventLoop
+    }
+    
+    public var information: InformationSet {
+        InformationSet(headers.map { key, rawValue in
+            AnyHTTPInformation(key: key, rawValue: rawValue)
+        })
+            .merge(with: [
+                InformationEntryHTTPVersion(key: .shared, rawValue: version)
+            ])
+            .merge(with: self.loggingMetadata)
+    }
+    
+    public var description: String {
+        "<\(Self.self) \(version.description) \(method.rawValue) \(url.stringValue)>"
+    }
+    
+    public var debugDescription: String {
+        self.description
+    }
+    
+    
+    var loggingMetadata: [LoggingMetadataInformation] {
+         [
+            LoggingMetadataInformation(key: .init("ApodiniNetworkingRequestDescription"), rawValue: .string(self.description)),
+            //LoggingMetadataInformation(key: .init("HTTPBody"), rawValue: .string(self.bodyData.count < 32_768 ? String(decoding: self.bodyData, as: UTF8.self) : "\(String(decoding: self.bodyData, as: UTF8.self).prefix(32_715))... (further bytes omitted since HTTP body too large!")),
```

Comment:
any reason those got commented out?

Reply:
Added some of these (HTTP version, content type) back.
The main issue here is that we simply don't have access to some of this information anymore. Or, rather, we can't access it the same way as Vapor would let us. For example, we now have a much clearer separation between buffer- and stream-based request bodies, which means that we can't just "get some bytes" to display in the logging metadata.